atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceMigrators.java

```diff
@@ -43,11 +46,14 @@ private KeyValueServiceMigrators() {
          * certain tables to always be in the legacy DB KVS (because that one supports
          * putUnlessExists), and those tables cannot and should not be migrated. Also, special
          * tables that are not controlled by the transaction table should not be
-         * migrated.
+         * migrated. Lastly, we never migrate any checkpoints from previous KVS migration attempts,
+         * since that would corrupt the current migration. Since the namespace might have changed, we
+         * remove all table names that match the internal checkpoint table name.
          */
         Set<TableReference> tableNames = Sets.newHashSet(kvs.getAllTableNames());
         tableNames.removeAll(AtlasDbConstants.hiddenTables);
         tableNames.removeAll(unmigratableTables);
+        tableNames.removeIf(tableRef -> tableRef.getTablename().equals(CHECKPOINT_TABLE_NAME));
```

Comment:
ohhh, this namespace thing keeps biting me 😅 , never mind me

Reply:
If all tables in any namespace that match this name should be considered as the special checkpoint table, it might be good to prevent normal tables from having this name (or can we change it so that this table also works like other special tables where it always has an empty namespace?).